src/cloudant/views.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+"""
+_views_
+
+Utilities for handling design docs and the resulting views they create
+
+"""
+from .document import CloudantDocument
+
+
+class Code(str):
+    """
+    _Code_
+
+    string derived object that allows us to wrap/manipulate javascript blobs
+
+    """
+    def __init__(self, s):
+        super(Code, self).__init__(s)
+
+
+class View(dict):
+    """
+    Dictionary based object representing a view, exposing the map, reduce
+    functions as attributes and supporting query/data access via the view
+
+    """
+    def __init__(self, view_name, map_func=None, reduce_func=None):
+        super(View, self).__init__()
+        self.view_name = view_name
+        self[self.view_name] = {}
```

Comment:
Its how the view dictionaries are structured in the design doc. Each view is defined as { view_name: { "map"..., "reduce"...} } so this nesting makes it ~trivial to translate the view object to/from JSON, rather than having to mangle it.